src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -44,26 +51,25 @@ public CourseScoreCalculationService(StudentParticipationRepository studentParti
         this.plagiarismCaseRepository = plagiarismCaseRepository;
     }
 
+    private record MaxAndReachablePoints(double maxPoints, double reachablePoints) {
+    }
+
     /**
-     * Calculates max and reachable max points for the given course and the student scores for the given student ids
-     * and takes the effects of related plagiarism verdicts on the grade into account. Implementation is adapted from course-score-calculation.service.ts.
-     * <p>
-     * If there is a single student id in studentIds, the student id will be filtered in the database as an optimization
-     * wherever possible.
+     * Calculates max and reachable max points for the given exercises.
```

Comment:
Do we somewhere have an explanation what exactly the difference between max and reachable max is? Please either explaint that here or link that explanation.

Reply:
The explanation is distributed across this class. I consolidated it in this JavaDoc.